evo_researcher/benchmark/benchmark.py

```diff
@@ -47,12 +56,32 @@ def __init__(
         self.metric_fns.update(predefined_metric_fns)
 
     def add_prediction(
-        self, agent: AbstractBenchmarkedAgent, prediction: PredictionResult
+        self,
+        agent: AbstractBenchmarkedAgent,
+        prediction: PredictionResult,
+        market_question: str,
     ):
-        self.predictions[agent.agent_name].append(prediction)
+        self.predictions.add_prediction(
+            agent_name=agent.agent_name,
+            question=market_question,
+            prediction=prediction,
+        )
+
+    def get_prediction(self, agent_name: str, question: str) -> PredictionResult:
+        return self.predictions.get_prediction(agent_name=agent_name, question=question)
 
     def run_agents(self):
         for agent in self.registered_agents:
+            if self.cache_path:
+                markets_to_run = [
+                    m
+                    for m in self.markets
+                    if not self.predictions.has_market(
+                        agent_name=agent.agent_name, question=m.question
+                    )
+                ]
+            else:
+                markets_to_run = self.markets
```

Comment:
It seems that `self.predictions` is always initialized, even if empty: 

```
if self.cache_path and os.path.exists(self.cache_path):
            self.predictions = PredictionsCache.load(path=self.cache_path)
else:
            self.predictions = PredictionsCache(predictions={})
```

Is this if/else necessary? Seems like 

```
 markets_to_run = [
                    m
                    for m in self.markets
                    if not self.predictions.has_market(
                        agent_name=agent.agent_name, question=m.question
                    )
                ]
```

Should be enough.

Reply:
Yes, you're right! Fixed.